front/lib/resources/data_source_view_resource.ts

```diff
@@ -265,14 +258,24 @@ export class DataSourceViewResource extends ResourceWithVault<DataSourceViewMode
     return isResourceSId("data_source_view", sId);
   }
 
+  getUsagesByAgents = async (auth: Authenticator) => {
+    return getDataSourceViewUsage({ auth, dataSourceView: this.toJSON() });
+  };
+
   // Serialization.
 
   toJSON(): DataSourceViewType {
     return {
+      id: this.id,
+      sId: this.sId,
       createdAt: this.createdAt.getTime(),
       parentsIn: this.parentsIn,
-      sId: this.sId,
       updatedAt: this.updatedAt.getTime(),
+      connectorId: this.dataSource.connectorId,
+      connectorProvider: this.dataSource.connectorProvider,
+      name: this.dataSource.name,
+      description: this.dataSource.description,
+      dustAPIProjectId: this.dataSource.dustAPIProjectId,
```

Comment:
Why do we need all those fields?

Reply:
it's to support DataSourceOrViewInfo and have consistent data between datasources and views. But it's not really needed in the datasourceview json directly, and we could keep the logic of merging data in getDataSourceViewInfo.